proposals/solid-oidc/primer.bs

```diff
@@ -0,0 +1,745 @@
+
+For example, our client id is `https://decentphotos.com/webid#this` and our client secret is `thisIsASecret`, so the full string would be `https://decentphotos.com/webid#this:thisIsASecret` which is `aHR0cHM6Ly9kZWNlbnRwaG90b3MuY29tL3dlYmlkI3RoaXM6dGhpc0lzQVNlY3JldA==` when base64 encoded.
+
+NOTE: Based on the spec, it is currently unknown how an app with an app WebId gets its client secret.
+
+ #### 16. Token requst with code and code verifier
+
+Now, we have everything we need to make an auth request. No need to redirect the web browser for this one. We only need to make an AJAX request to the `token` enpoint as defined in the OS's `openid-configuration` file, in our case `https://secureauth.example/token`
+
+```
+POST https://secureauth.example/token
+Headers: {
+  "Authorization": "Basic aHR0cHM6Ly9kZWNlbnRwaG90b3MuY29tL3dlYmlkI3RoaXM6dGhpc0lzQVNlY3JldA==",
+  "DPoP": "eyJhbGciOiJFUzI1NiIsInR5cCI6ImRwb3Arand0IiwiandrIjp7Imt0eSI6IkVDIiwia2lkIjoiZkJ1STExTkdGbTQ4Vlp6RzNGMjVDOVJmMXYtaGdEakVnV2pEQ1BrdV9pVSIsInVzZSI6InNpZyIsImFsZyI6IkVDIiwiY3J2IjoiUC0yNTYiLCJ4IjoiOWxlT2gxeF9IWkhzVkNScDcyQzVpR01jek1nUnpDUFBjNjBoWldfSFlLMCIsInkiOiJqOVVYcnRjUzRLVzBIYmVteW1vRWlMXzZ1cko0TFFHZXJQZXVNaFNEaV80In19.eyJodHUiOiJodHRwczovL3NlY3VyZWF1dGguZXhhbXBsZS90b2tlbiIsImh0bSI6InBvc3QiLCJqdGkiOiI0YmEzZTllZi1lOThkLTQ2NDQtOTg3OC03MTYwZmE3ZDNlYjgiLCJpYXQiOjE2MDMzMDYxMjgsImV4cCI6MTYwMzMwOTcyOH0.2lbgLoRCkj0MsDc9BpquoaYuq0-XwRf_URdXru2JKrVzaWUqQfyKRK76_sQ0aJyVwavM3pPswLlHq2r9032O7Q",
+  "content-type": "application/x-www-form-urlencoded"
+}
+Body:
+  grant_type=authorization_code&
+  code_verifier=JXPOuToEB7&
+  code=m-OrTPHdRsm8W_e9P0J2Bt&
+  redirect_uri=https%3A%2F%2Fdecentphotos.example%2Fcallback&
+  client_id=https%3A%2F%2Fdecentphotos.example%2Fwebid%23this
+```
+
+ - `headers.Authorization: "Basic aHR0cHM6L..."`: The basic authorization we made in the previous step. Be sure to include the word "Basic" followed by a space in this header.
+ - `headers.DPoP: "eyJhbGciOiJFUz..."`: The DPoP token we generated before. This will tell the OS what the client's public key is.
+ - `headers.content-type: "application/x-www-form-urlencoded"`: Sets to body's content type to `application/x-www-form-urlencoded`. Some OSs will accept other content types like `application/json` but they all must access urlencoded content types, so it's safest to use that.
+ - `body.grant_type=authorization_code`: Tells the OS that we are doing the authorization code flow.
+ - `body.code_verifier=JXPOuToEB7`: Our code verifier that we stored in local storage
+ - `body.code=m-OrTPHdRsm8W_e9P0J2Bt`: The code that we received from the OS upon redirect
+ - `body.redirect_uri`: The app's redirect url. In this case, this isn't needed because we're doing an AJAX request.
+ - `body.client_id=https%3A%2F%2Fdecentphotos.example%2Fwebid%23this`: The app's client id.
+
+Once this request is completed decentphotos can remove the code verifier from local storage.
+
+#### 17. Validate Client Id and Secret
```

Comment:
Section 17 is only relevant for static and dynamic OIDC registration. For WebID-based client id (such as with this example), it is not relevant.